query_execution/QueryContext.cpp

```diff
@@ -58,7 +59,8 @@ QueryContext::QueryContext(const serialization::QueryContext &proto,
                            const CatalogDatabaseLite &database,
                            StorageManager *storage_manager,
                            const tmb::client_id scheduler_client_id,
-                           tmb::MessageBus *bus) {
+                           tmb::MessageBus *bus)
+    : database_(static_cast<const CatalogDatabase&>(database)) {
```

Comment:
That would give us partial information - we will miss information about blocks which get full during the operator execution. Right now, the way we find out the memory occupied by a temporary relation is simply by estimating size of each block and multiplying that with the number of blocks in the temporary relation.